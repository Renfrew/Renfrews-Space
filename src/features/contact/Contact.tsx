import React, { useReducer, ReactElement } from 'react';
import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';

import {
  Button,
  Container,
  Grid,
  Hidden,
  TextField,
  Typography,
} from '@material-ui/core';

import { ContactMeType, thunk, reducer } from './Reducer';

const useStyles = makeStyles(({ breakpoints, spacing }: Theme) =>
  createStyles({
    header: {
      textAlign: 'center',
      marginTop: spacing(10),
      marginBottom: spacing(8),
    },
    parent: {
      justifyContent: 'center',
      textAlign: 'center',
      marginLeft: 'auto',
      marginRight: 'auto',
      maxWidth: breakpoints.values.md,
    },
    textField: {
      [breakpoints.up('md')]: {
        width: '75%',
      },
    },
    sendButton: {
      textAlign: 'center',
      marginTop: spacing(4),
    },
    leftContainer: {
      [breakpoints.up('md')]: {
        paddingRight: spacing(1),
        textAlign: 'right',
        '& > div': {
          width: '70%',
        },
      },
    },
    rightContainer: {
      [breakpoints.up('md')]: {
        paddingLeft: spacing(1),
        textAlign: 'left',
        '& > div': {
          width: '78%',
        },
      },
    },
  })
);

const initialState = {
  name: '',
  email: '',
  website: '',
  message: '',
};

const Contact = (): ReactElement => {
  const classes = useStyles();

  const [{ name, email, website, message }, dispatch] = useReducer(
    reducer,
    initialState
  );

  function onChange(event: React.ChangeEvent<HTMLInputElement>) {
    dispatch(thunk(event.target.id, event.target.value));
  }
  function onSubmitClick(
    event: React.MouseEvent<HTMLButtonElement, MouseEvent>
  ) {
    event.preventDefault();
  }

  return (
    <Container component="main" disableGutters maxWidth={false}>
      <Typography variant="h2" className={classes.header}>
        Contact Me
      </Typography>
      <Grid
        container
        component="form"
        noValidate
        autoComplete="off"
        className={classes.parent}
      >
        <Grid item xs={10} sm={7} md={5} className={classes.leftContainer}>
          <TextField
            required
            fullWidth
            margin="normal"
            id={ContactMeType.Name}
            label="Name"
            value={name}
            onChange={onChange}
          />
        </Grid>
        <Grid item xs={10} sm={7} md={7} className={classes.rightContainer}>
          <TextField
            required
            fullWidth
            margin="normal"
            id={ContactMeType.Email}
            label="Email"
            value={email}
            onChange={onChange}
          />
        </Grid>
        <Hidden mdDown>
          <Grid item md={5} className={classes.leftContainer}>
            <TextField
              fullWidth
              margin="normal"
              id={ContactMeType.Website}
              label="Website"
              value={website}
              onChange={onChange}
            />
          </Grid>
        </Hidden>
        <Grid item xs={10} sm={7} md={7} className={classes.rightContainer}>
          <TextField
            fullWidth
            margin="normal"
            id={ContactMeType.Website}
            label="Website"
            value={website}
            onChange={onChange}
          />
        </Grid>
        <Grid item xs={10} sm={7} md={12}>
          <TextField
            fullWidth
            margin="normal"
            className={classes.textField}
            id={ContactMeType.Message}
            label="Type your message here"
            multiline
            rows={5}
            value={message}
            onChange={onChange}
          />
        </Grid>
        <Grid item xs={12} sm={7} className={classes.sendButton}>
          <Button
            onClick={onSubmitClick}
            variant="contained"
            color="secondary"
            size="large"
          >
            SEND
          </Button>
        </Grid>
      </Grid>
    </Container>
  );
};

export default Contact;
