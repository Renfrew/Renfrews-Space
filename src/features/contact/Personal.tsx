import React, { ReactElement } from 'react';
import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';

import { Container, Grid, Typography } from '@material-ui/core';

const useStyles = makeStyles(({ breakpoints, spacing }: Theme) =>
  createStyles({
    header: {
      textAlign: 'center',
      marginTop: spacing(3),
      marginBottom: spacing(1),
      [breakpoints.up('md')]: {
        marginTop: spacing(10),
        marginBottom: spacing(3),
      },
    },
    line: {
      justifyContent: 'center',
      marginLeft: 'auto',
      marginRight: 'auto',
      maxWidth: breakpoints.values.md,
      '& div': {
        marginLeft: spacing(2),
      },
    },
  })
);

interface Props {
  label: string;
  data: string | number;
}

const info: { label: string; value: string | number }[] = [
  { label: 'Phone', value: '(226) 929 -5495' },
  {
    label: 'Email1:',
    value: 'l23chen@uwaterloo.ca',
  },
  {
    label: 'Email2:',
    value: 'renfrew.liangchen@gmail.com',
  },
  {
    label: 'Address',
    value: 'Hamilton, ON. (can move)',
  },
];

const PersonalInfo = (): ReactElement => {
  const classes = useStyles();

  function Line({ label, data }: Props) {
    return (
      <Grid container className={classes.line}>
        <Grid item>
          <Typography>{label}</Typography>
        </Grid>
        <Grid item>
          <Typography>{data}</Typography>
        </Grid>
      </Grid>
    );
  }

  return (
    <Container disableGutters maxWidth={false}>
      <Typography variant="h4" className={classes.header}>
        Personal Information
      </Typography>
      {info.map((el) => (
        <Line key={el.label} label={el.label} data={el.value} />
      ))}
    </Container>
  );
};

export default PersonalInfo;
