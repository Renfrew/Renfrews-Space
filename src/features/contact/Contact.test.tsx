import React from 'react';
import renderer from 'react-test-renderer';
import Contact from './Contact';
import Personal from './Personal';
import SendMesssage from './SendMessage';

describe('Testing the Contact page', () => {
  it('The Personal module renders correctly', () => {
    const tree = renderer.create(<Personal />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('The send message  module renders correctly', () => {
    const tree = renderer.create(<SendMesssage />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('The entired contact module renders correctly', () => {
    const tree = renderer.create(<Contact />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
