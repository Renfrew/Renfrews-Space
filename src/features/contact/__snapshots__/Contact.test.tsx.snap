// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Contact page The Personal module renders correctly 1`] = `
<div
  className="MuiContainer-root MuiContainer-disableGutters css-q7bdj4-MuiContainer-root"
>
  <h4
    className="MuiTypography-root MuiTypography-h4 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial makeStyles-header-1 css-5lbw0b-MuiTypography-root"
  >
    Personal Information
  </h4>
  <div
    className="MuiGrid-root MuiGrid-container makeStyles-line-2 css-15fev7r-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item css-10texkh-MuiGrid-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial css-ahj2mt-MuiTypography-root"
      >
        Phone
      </p>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item css-10texkh-MuiGrid-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial css-ahj2mt-MuiTypography-root"
      >
        (226) 929 -5495
      </p>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container makeStyles-line-2 css-15fev7r-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item css-10texkh-MuiGrid-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial css-ahj2mt-MuiTypography-root"
      >
        Email1:
      </p>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item css-10texkh-MuiGrid-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial css-ahj2mt-MuiTypography-root"
      >
        l23chen@uwaterloo.ca
      </p>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container makeStyles-line-2 css-15fev7r-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item css-10texkh-MuiGrid-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial css-ahj2mt-MuiTypography-root"
      >
        Email2:
      </p>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item css-10texkh-MuiGrid-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial css-ahj2mt-MuiTypography-root"
      >
        renfrew.liangchen@gmail.com
      </p>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container makeStyles-line-2 css-15fev7r-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item css-10texkh-MuiGrid-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial css-ahj2mt-MuiTypography-root"
      >
        Address
      </p>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item css-10texkh-MuiGrid-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial css-ahj2mt-MuiTypography-root"
      >
        Hamilton, ON. (willing to move)
      </p>
    </div>
  </div>
</div>
`;

exports[`Testing the Contact page The entired contact module renders correctly 1`] = `
<main
  className="MuiContainer-root MuiContainer-disableGutters makeStyles-container-13 css-q7bdj4-MuiContainer-root"
>
  <h2
    className="MuiTypography-root MuiTypography-h2 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial makeStyles-header-14 css-1sra7t5-MuiTypography-root"
  >
    Contact Me
  </h2>
  <div
    className="MuiContainer-root MuiContainer-disableGutters css-q7bdj4-MuiContainer-root"
  >
    <h4
      className="MuiTypography-root MuiTypography-h4 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial makeStyles-header-1 css-5lbw0b-MuiTypography-root"
    >
      Personal Information
    </h4>
    <div
      className="MuiGrid-root MuiGrid-container makeStyles-line-2 css-15fev7r-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item css-10texkh-MuiGrid-root"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial css-ahj2mt-MuiTypography-root"
        >
          Phone
        </p>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item css-10texkh-MuiGrid-root"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial css-ahj2mt-MuiTypography-root"
        >
          (226) 929 -5495
        </p>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container makeStyles-line-2 css-15fev7r-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item css-10texkh-MuiGrid-root"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial css-ahj2mt-MuiTypography-root"
        >
          Email1:
        </p>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item css-10texkh-MuiGrid-root"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial css-ahj2mt-MuiTypography-root"
        >
          l23chen@uwaterloo.ca
        </p>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container makeStyles-line-2 css-15fev7r-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item css-10texkh-MuiGrid-root"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial css-ahj2mt-MuiTypography-root"
        >
          Email2:
        </p>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item css-10texkh-MuiGrid-root"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial css-ahj2mt-MuiTypography-root"
        >
          renfrew.liangchen@gmail.com
        </p>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container makeStyles-line-2 css-15fev7r-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item css-10texkh-MuiGrid-root"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial css-ahj2mt-MuiTypography-root"
        >
          Address
        </p>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item css-10texkh-MuiGrid-root"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial css-ahj2mt-MuiTypography-root"
        >
          Hamilton, ON. (willing to move)
        </p>
      </div>
    </div>
  </div>
  <div
    className="MuiContainer-root MuiContainer-disableGutters css-q7bdj4-MuiContainer-root"
  >
    <h4
      className="MuiTypography-root MuiTypography-h4 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial makeStyles-header-3 css-5lbw0b-MuiTypography-root"
    >
      Send Message
    </h4>
    <form
      autoComplete="off"
      className="MuiGrid-root MuiGrid-container makeStyles-parent-4 css-15fev7r-MuiGrid-root"
      onSubmit={[Function]}
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-sm-7 MuiGrid-grid-md-5 makeStyles-leftContainer-7 css-1gqg8cn-MuiGrid-root"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
            data-shrink={false}
            htmlFor="name"
            id="name-label"
          >
            Name
            <span
              aria-hidden={true}
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-luel6v-MuiInputBase-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input css-gga68z-MuiInputBase-input"
              disabled={false}
              id="name"
              name="name"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-11"
              >
                <span>
                  Name
                   *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-sm-7 MuiGrid-grid-md-7 makeStyles-rightContainer-8 css-roj13u-MuiGrid-root"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
            data-shrink={false}
            htmlFor="email"
            id="email-label"
          >
            Email
            <span
              aria-hidden={true}
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-luel6v-MuiInputBase-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input css-gga68z-MuiInputBase-input"
              disabled={false}
              id="email"
              name="email"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="email"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-11"
              >
                <span>
                  Email
                   *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-sm-7 MuiGrid-grid-md-7 makeStyles-rightContainer-8 css-roj13u-MuiGrid-root"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
            htmlFor="website"
            id="website-label"
          >
            Website
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-luel6v-MuiInputBase-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input css-gga68z-MuiInputBase-input"
              disabled={false}
              id="website"
              name="website"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="url"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-11"
              >
                <span>
                  Website
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-sm-7 MuiGrid-grid-md-12 css-nvm0ql-MuiGrid-root"
      >
        <div
          className="MuiFormControl-root MuiTextField-root makeStyles-textField-5 MuiFormControl-marginNormal MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
            htmlFor="message"
            id="message-label"
          >
            Type your message here
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiOutlinedInput-multiline MuiInputBase-multiline css-mad0u6-MuiInputBase-root"
            onClick={[Function]}
          >
            <textarea
              aria-invalid={false}
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input MuiOutlinedInput-inputMultiline MuiInputBase-inputMultiline css-kjyl2c-MuiInputBase-input"
              disabled={false}
              id="message"
              name="message"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              rows={5}
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-11"
              >
                <span>
                  Type your message here
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-7 makeStyles-sendButton-6 css-ic93ua-MuiGrid-root"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeLarge MuiButton-containedSizeLarge css-1olissi-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          <span
            className="MuiButton-label css-8xplcm-MuiButton-label"
          >
            SEND
          </span>
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </form>
  </div>
</main>
`;

exports[`Testing the Contact page The send message  module renders correctly 1`] = `
<div
  className="MuiContainer-root MuiContainer-disableGutters css-q7bdj4-MuiContainer-root"
>
  <h4
    className="MuiTypography-root MuiTypography-h4 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial makeStyles-header-3 css-5lbw0b-MuiTypography-root"
  >
    Send Message
  </h4>
  <form
    autoComplete="off"
    className="MuiGrid-root MuiGrid-container makeStyles-parent-4 css-15fev7r-MuiGrid-root"
    onSubmit={[Function]}
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-sm-7 MuiGrid-grid-md-5 makeStyles-leftContainer-7 css-1gqg8cn-MuiGrid-root"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink={false}
          htmlFor="name"
          id="name-label"
        >
          Name
          <span
            aria-hidden={true}
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-luel6v-MuiInputBase-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiOutlinedInput-input MuiInputBase-input css-gga68z-MuiInputBase-input"
            disabled={false}
            id="name"
            name="name"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-11"
            >
              <span>
                Name
                 *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-sm-7 MuiGrid-grid-md-7 makeStyles-rightContainer-8 css-roj13u-MuiGrid-root"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink={false}
          htmlFor="email"
          id="email-label"
        >
          Email
          <span
            aria-hidden={true}
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-luel6v-MuiInputBase-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiOutlinedInput-input MuiInputBase-input css-gga68z-MuiInputBase-input"
            disabled={false}
            id="email"
            name="email"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="email"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-11"
            >
              <span>
                Email
                 *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-sm-7 MuiGrid-grid-md-7 makeStyles-rightContainer-8 css-roj13u-MuiGrid-root"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink={false}
          htmlFor="website"
          id="website-label"
        >
          Website
        </label>
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-luel6v-MuiInputBase-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiOutlinedInput-input MuiInputBase-input css-gga68z-MuiInputBase-input"
            disabled={false}
            id="website"
            name="website"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="url"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-11"
            >
              <span>
                Website
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-sm-7 MuiGrid-grid-md-12 css-nvm0ql-MuiGrid-root"
    >
      <div
        className="MuiFormControl-root MuiTextField-root makeStyles-textField-5 MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink={false}
          htmlFor="message"
          id="message-label"
        >
          Type your message here
        </label>
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiOutlinedInput-multiline MuiInputBase-multiline css-mad0u6-MuiInputBase-root"
          onClick={[Function]}
        >
          <textarea
            aria-invalid={false}
            autoFocus={false}
            className="MuiOutlinedInput-input MuiInputBase-input MuiOutlinedInput-inputMultiline MuiInputBase-inputMultiline css-kjyl2c-MuiInputBase-input"
            disabled={false}
            id="message"
            name="message"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            rows={5}
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-11"
            >
              <span>
                Type your message here
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-7 makeStyles-sendButton-6 css-ic93ua-MuiGrid-root"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeLarge MuiButton-containedSizeLarge css-1olissi-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label css-8xplcm-MuiButton-label"
        >
          SEND
        </span>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </form>
</div>
`;
